src/WebJobs.Script/Scale/FunctionsScaleManager.cs

```diff
@@ -84,15 +116,69 @@ public virtual async Task<ScaleStatusResult> GetScaleStatusAsync(ScaleStatusCont
                 // this can happen if the host is offline
             }
 
-            var vote = GetAggregateScaleVote(votes, context, _logger);
+            ScaleVote vote = GetAggregateScaleVote(votes.Union(targetScaleVotes), context, _logger);
+            return vote;
+        }
 
-            return new ScaleStatusResult
+        private async Task<IEnumerable<TargetScalerVote>> GetTargetScalersResult(ScaleStatusContext context, IEnumerable<ITargetScaler> targetScalersToProcess)
+        {
+            List<TargetScalerVote> targetScaleVotes = new List<TargetScalerVote>();
+
+            if (targetScalersToProcess.Any())
             {
-                Vote = vote
-            };
+                _logger.LogDebug($"{targetScalersToProcess.Count()} target scalers to sample");
+                HostConcurrencySnapshot snapshot = null;
+                try
+                {
+                    snapshot = await _concurrencyStatusRepository.ReadAsync(CancellationToken.None);
+                }
+                catch (Exception exc) when (!exc.IsFatal())
+                {
+                    _logger.LogError(exc, $"Failed to read concurrency status repository");
+                }
+
+                foreach (var targetScaler in targetScalersToProcess)
+                {
+                    try
+                    {
+                        TargetScalerContext targetScaleStatusContext = new TargetScalerContext();
+                        if (snapshot != null)
+                        {
+                            if (snapshot.FunctionSnapshots.TryGetValue(targetScaler.TargetScalerDescriptor.FunctionId, out var functionSnapshot))
+                            {
+                                targetScaleStatusContext.InstanceConcurrency = functionSnapshot.Concurrency;
+                                _logger.LogDebug($"Snapshot dynamic concurrency for target scaler '{targetScaler.TargetScalerDescriptor.FunctionId}' is '{functionSnapshot.Concurrency}'");
+                            }
+                        }
+                        TargetScalerResult result = await targetScaler.GetScaleResultAsync(targetScaleStatusContext);
```

Comment:
nit: please space this code out for readability. E.g. space on lines 153, 155, 164. Basically space apart logical blocks